mne/epochs.py

```diff
@@ -2064,6 +2064,280 @@ def _drop_log_stats(drop_log, ignore=('IGNORED',)):
+    """Generate a default set of `~mne.Epochs` metadata.
+
+    This function generates metadata based on all events falling into the time
+    interval ``[tmin, tmax]``. Only events specified in ``event_id`` will be
+    considered.
+
+    The function will also return a subset of the events array that corresponds
+    to the generated metadata, so both the metadata and the events may directly
+    be used to create `~mne.Epochs`.
+
+    You may specify which events are the time-locked events that will later
+    be used to create `~mne.Epochs` via the ``time_locked_events`` parameter.
+
+    By default, the time of the **first occurrence** of each event is extracted
+    from every time period. To keep the **last occurrence** of a certain
+    event instead, pass ``keep_last``.
+
+    Lastly, you may specify hierarchical event descriptors (HED) in
+    ``keep_first`` and ``keep_last`` to keep the time of the first and last
+    occurrence, respectively, of an entire group of different events.
+    For example, if you have the following hierarchical event names:
+    ``stim/a``, ``stim/b`` and pass ``keep_first='stim'``, the time of
+    **either** of the two events will be extracted, depending on which of the
+    two occurred first in the current time period.
+
+    Parameters
+    ----------
+    events : array, shape (m, 3)
+        The events array.
+    event_id : dict
+        The mapping from event names (keys) to event IDs (values). This
+        function assumes that only event IDs listed here will be used to
+        create `~mne.Epochs`, that is, the number of rows of the generated
+        metadata will equal the number of events with the specified IDs.
+        Moreover, metadata will only be generated for the events listed here.
+    tmin : float
+        Start of the time interval for event extraction in seconds, relative to
+        time-locked event of the respective epoch. This will typically be the
+        same ``tmin`` you'll pass to `~mne.Epochs`, but it may also be
+        different to include more or fewer events in the metadata.
+    tmax : float
+        Start of the time interval for event extraction in seconds, relative to
+        time-locked event of the respective epoch. This will typically be the
+        same ``tmax`` you'll pass to `~mne.Epochs`, but it may also be
+        different to include more or fewer events in the metadata.
+    sfreq : float
+        The sampling frequency during data acquisiton.
+    time_locked_events : list of str | str | None
+        This can be a subset of event names (keys of ``event_id``). Metadata
+        rows will only be created for the specified event(s), instead of for
+        all events present in ``event_id``. If ``None`` (default), create
+        one row for each event in ``event_id``.
+    keep_first : str | list of str | None
+        Specify hierarchical event descriptors(s) (HED) of events from which
```

Comment:
do these *have to be* HEDs? I can imagine a case of multiple buttonpresses where I only want the first one (but there is no `button/left` or `button/right`, just one `button`)

Reply:
I have tried to address this in 9560140

Is you probably can tell, I'm struggling a little due to the complex nature of all of this 😅
But is it clearer to you now?